apps/src/music/views/TimelineSimple2Events.jsx

```diff
@@ -0,0 +1,142 @@
+import PropTypes from 'prop-types';
+import React, {useContext} from 'react';
+import {PlayerUtilsContext} from '../context';
+import TimelineElement from './TimelineElement';
+
+/**
+ * Renders timeline events for the simple2 model.
+ */
+const TimelineSimple2Events = ({
+  currentPlayheadPosition,
+  barWidth,
+  eventVerticalSpace,
+  getEventHeight
+}) => {
+  const playerUtils = useContext(PlayerUtilsContext);
+  const soundEvents = playerUtils.getSoundEvents();
+
+  const getVerticalOffsetForEventId = id => {
+    return (
+      getUniqueIndexForEventId(id) * getEventHeight(currentUniqueSounds.length)
+    );
+  };
+
+  const getUniqueIndexForEventId = id => {
+    return currentUniqueSounds.indexOf(id);
+  };
+
+  // Generate a list of unique sounds, with uniqueness being a combination of
+  // the function name and the sound ID.
+  // Let's cache the value of currentUniqueSounds so that the various helpers
+  // we call during render don't need to recalculate it.  This also ensures
+  // that we recalculate unique sounds, even when there are no entries to
+  // render.
+  const currentUniqueSounds = [];
```

Comment:
If you want to cache some or all of this calculation, I think you can use the `useMemo` hook here (I did something similar in `TimelineTrackEvents`). That way this will only run whenever `soundEvents` changes

Reply:
Oh cool.  I will learn more and hopefully do a follow-up with this capability.